### PR TITLE
Detach app service from dev builds

### PR DESCRIPTION
**Pull request by FirePing32**
Nest app is now detached from dev builds and runs separately.

fixes:#148

CC: @yuvrajsab

### CODE REVIEW

**Comment by FirePing32** [comment]
Why is GitGuardian failing ?

**Comment by FirePing32** [comment]
Why is GitGuardian failing ?
@yuvrajsab @ChakshuGautam

**Comment by yuvrajsab** [comment]
@FirePing32, It checks for any secret in the code base since .env.copy file contains a dummy `HASURA_ADMIN_SECRET` secret it is giving this warning. you can ignore this.

File: README.md [hunk 1]
```diff
@@ -19,11 +19,23 @@ The project is built on a plugin model, which ensures customizability and wide a
...
+### Development
+
 ```bash
 # Specify the local compose file
 $ docker-compose -f docker-compose.local.yml up -d --build
 
-# For production
+# Installing app dependencies
+$ yarn install
+
```
Comment (yuvrajsab) [inline README.md:35]: add a step to run Prisma migrations.
```bash
yarn prisma migrate dev -n init
```

File: README.md [hunk 2]
```diff
@@ -19,11 +19,23 @@ The project is built on a plugin model, which ensures customizability and wide a
...
+# Start in watch mode
+$ yarn start:dev
+```
+
+### Production
+
+```bash
+# Production build includes the app service
 $ docker-compose up -d --build
 ```
```
Comment (yuvrajsab) [inline README.md:66]: add a step here to run migrations in the docker container.
```bash
# go inside the container
docker-compose exec app bash

# run migration
npx prisma migrate dev

# exit out of container
exit
```

File: README.md [hunk 3]
```diff
@@ -19,11 +19,23 @@ The project is built on a plugin model, which ensures customizability and wide a
...
+# Installing app dependencies
+$ yarn install
+
+# Start in watch mode
+$ yarn start:dev
+```
+
+### Production
+
+```bash
```
Comment (yuvrajsab) [inline README.md:50]: add a step for creating env file.

File: README.md [hunk 4]
```diff
@@ -19,11 +19,23 @@ The project is built on a plugin model, which ensures customizability and wide a
 $ cp .env.copy .env
 ```
 
+### Development
+
 ```bash
 # Specify the local compose file
 $ docker-compose -f docker-compose.local.yml up -d --build
```
Comment (yuvrajsab) [inline README.md]: can remove `--build` flag from here.

**Review by yuvrajsab — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (FirePing32) [reply README.md:50]: It's a common step for both. That's why it is included at the top, and separate subheadings have been created for development and production.

File: README.md (hunk 2 above)
Comment (FirePing32) [reply README.md:66]: Should we instead add these in the Dockerfile ?

Comment (yuvrajsab) [reply README.md:66]: no, we don't want to run migrations every time we build a new docker image.

File: README.md (hunk 3 above)
Comment (yuvrajsab) [reply README.md:50]: it's could be confusing, would recommend you add this step to both envs.

File: README.md [hunk 5]
```diff
@@ -14,17 +14,42 @@ The project is built on a plugin model, which ensures customizability and wide a
...
+# Start in watch mode
+$ yarn start:dev
 ```
 
+### Production
+
 ```bash
-# Specify the local compose file
-$ docker-compose -f docker-compose.local.yml up -d --build
+# Copy env file
```
Comment (yuvrajsab) [inline README.md]: add a comment something like "You need to edit .env according to your prod setup" ...

File: README.md [hunk 6]
```diff
@@ -14,17 +14,42 @@ The project is built on a plugin model, which ensures customizability and wide a
 
 ## Running the app
 
+### Development
+
```
Comment (yuvrajsab) [inline README.md:18]: add pre-requisites here that are node 16, yarn 1, docker and docker-compose.

File: README.md [hunk 7]
```diff
@@ -14,17 +14,42 @@ The project is built on a plugin model, which ensures customizability and wide a
...
+# Run Prisma migrations
+$ yarn prisma migrate dev -n init
+
+# Start in watch mode
+$ yarn start:dev
 ```
 
+### Production
+
```
Comment (yuvrajsab) [inline README.md:44]: add prerequisites here that are docker and docker-compose.

**Review by yuvrajsab — CHANGES_REQUESTED** [review]

**Review by yuvrajsab — APPROVED** [review]
LGTM!

**Comment by FirePing32** [comment]
@AnshulMalik Please merge.

**Comment by FirePing32** [comment]
@yuvrajsab Please merge.

**Comment by yuvrajsab** [comment]
@AnshulMalik please merge this.

**Pull request merged**